### PR TITLE
Database profanity - Johanna 

### PR DESCRIPTION
**Pull request by idsintehittapa**
Live project: https://database-profanity.herokuapp.com/profanities

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3,19 +3,41 @@ import bodyParser from 'body-parser'
...
+// using async function
+// reach the database, clear it, populate it
+if (process.env.RESET_DATABASE) {
+  const populateDatabase = async () => {
+    await Profanity.deleteMany()
+
+    profanityData.forEach(item => {
+      const newProfanity = new Profanity(item)
+      newProfanity.save(function (err, item) {
+        if (err) return console.error(err)
```
Comment (trangtara) [inline server.js:34]: console.error(err): this line of code is cool. New learning point for me.

File: server.js [hunk 2]
```diff
@@ -27,11 +49,46 @@ const app = express()
...
 
+// Middleware
+// Error message if server is down
+// has to be place before routs
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).send({ error: 'service unavailable' })
+  }
```
Comment (trangtara) [inline server.js:60]: Nice codes with error message.

File: server.js [hunk 3]
```diff
@@ -27,11 +49,46 @@ const app = express()
...
   res.send('Hello world')
 })
 
+// restful endpoint to entire database
+// has to use async and await since lines of code has to be
+// executed in specific order
+// queries 
+app.get('/profanities', async (req, res) => {
+  const allProfanities = await Profanity.find(req.query)
+  res.json(allProfanities)
```
Comment (trangtara) [inline server.js:74]: Short and simple. Nice job!

File: server.js [hunk 4]
```diff
@@ -27,11 +49,46 @@ const app = express()
...
+  try {
+    const singleProfanity = await Profanity.findOne({ id: req.params.id })
+    if (singleProfanity) {
+      return res.json(singleProfanity)
+    } else {
+      res.status(404).json({ error: 'profanity not found' })
+    }
+  } catch (err) {
+    res.status(400).json({ error: 'Invalid profanity ID' })
+  }
```
Comment (trangtara) [inline server.js:89]: Perfecto!

**Review by trangtara — APPROVED** [review]
Hi Johanna,
Thank you for making my code review so simple :-). You have completed the requirements of the Blue level which includes:
- 2 API endpoints in which one is the collection and one is for a single result
- Using the Mongoose model to export your own data into MongoDB database
- RESTful API
It's nice that you also covered error handling for the database disconnection and wrong/invalid Id.
That's all comment from me. Cheers!

**Review by KarolCislowski — APPROVED** [review]
I'm waiting for a possibility to add new records into your DB... Polish has $%^$ loads of profanities :D
Great Project 👍🏻

**Pull request closed without merging**